Style deal component

brkidder opened the pull request:
@cadecameron @d-rowe 

Major styling in deal.jsx
Added Price in an attempt to help declutter Price. Suggestions welcome.
Minor code added. Console.logs are purposeful for now.

Thank you for your time!

cadecameron commented on client/src/components/checkout.jsx:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import Deals from './deals';
 
-class App extends React.Component {
+class Checkout extends React.Component {
```
I think the rename makes sense. 👍

brkidder replied:
Yeah App was a bit too general

cadecameron commented on client/src/components/deal.jsx:
```diff
@@ -1,29 +1,198 @@
-      <span>{deal.salePercent + '% OFF'}</span>
-    </p>
-  </label>
+    {onSale ? (
```
Nice use of conditionals!

cadecameron commented on client/src/components/deal.jsx:
```diff
@@ -1,29 +1,198 @@
+
+const PercentWrap = styled(DealWrap)`
+  color: ${props => props.onSale ? "rgb(255, 0, 0)" : "rgb(52, 135, 0)"};
+  background-color: ${props => props.onSale ? "rgb(255, 210, 210)" : "rgb(234, 252, 222)"};
```
Did you manage to get this working in the end?

brkidder replied:
i had the red color coded rgb(255,0,0,0) i had too many 0's

cadecameron commented on client/src/components/deals.jsx:
```diff
@@ -1,19 +1,21 @@
-    const key = i + 1;
-    return (
-      <div key={key}>
+const Deals = ({ deals, onChange, selectedOption, onSale }) => (
```
🎉

d-rowe replied:
Nice use of destructuring in this refactor

d-rowe commented on client/src/components/deal.jsx:
```diff
@@ -1,29 +1,198 @@
+  }
+`;
+
+const PriceWrap = styled(DealWrap)`
```
Nice use of style extending

cadecameron replied:
Nice use of style extenders 👍

cadecameron commented on client/src/components/deals.jsx:
```diff
@@ -1,19 +1,21 @@
+const Deals = ({ deals, onChange, selectedOption, onSale }) => (
+  <form>
+    {deals.map((deal, i) => {
+      const key = i + 1;
```
Is the `i + 1` necessary? Is your code relying on this key property anywhere else?

brkidder replied:
I just didn't want a 0 based key, but no other reason than that.

d-rowe replied:
Was wondering this too. Could the unique deal title be used for key?

d-rowe commented:
Good work, this refactor makes everything look a lot more manageable.

brkidder commented:
Thank you! @cadecameron @d-rowe for taking a look first thing this morning.

brkidder commented:
Didn't mean to close it. Refactored the ordering of the style attributes. refactored key to something meaningful and refactored duplicate code from extending styles

d-rowe commented:
Just checked out the new changes, looks great to me.
